spec/services/app_store_client_spec.rb

```diff
@@ -88,6 +88,96 @@
     end
   end
 
+  describe '#put' do
+    let(:message) { { application_id: SecureRandom.uuid } }
+    let(:response) { double(:response, code:) }
+    let(:code) { 201 }
+    let(:username) { nil }
+
+    before do
+      allow(described_class).to receive(:put)
+        .and_return(response)
+      allow(ENV).to receive(:fetch).and_call_original
+      allow(ENV).to receive(:fetch).with('APP_STORE_USERNAME', nil)
+                                   .and_return(username)
+    end
+
+    context 'when APP_STORE_URL is present' do
+      before do
+        allow(ENV).to receive(:fetch).with('APP_STORE_URL', 'http://localhost:8000')
+                                     .and_return('http://some.url')
+      end
+
+      it 'posts the message to the specified URL' do
+        expect(described_class).to receive(:put)
+          .with('http://some.url/v1/application/',
+                body: message.to_json,
+                headers: { authorization: 'Bearer test-bearer-token' })
+
+        subject.put(message)
+      end
+
+      context 'when authentication is not configured' do
+        before do
+          allow(ENV).to receive(:fetch).with('APP_STORE_TENANT_ID', nil).and_return(nil)
+        end
+
+        it 'posts the message without headers' do
+          expect(described_class).to receive(:put)
+            .with('http://some.url/v1/application/',
+                  body: message.to_json)
+
+          subject.put(message)
+        end
+      end
+    end
+
+    context 'when APP_STORE_URL is not present' do
+      it 'posts the message to default localhost url' do
+        expect(described_class).to receive(:put)
+          .with('http://localhost:8000/v1/application/',
+                body: message.to_json,
+                headers: { authorization: 'Bearer test-bearer-token' })
+
+        subject.put(message)
+      end
+    end
+
+    context 'when APP_STORE_USERNAME is present' do
```

Comment:
This test is junk - there is no APP_STORE_USERNAME or APP_STORE_PASSWORD - and I've a horrible feeling that if you `git blame` this file you'll see a context like this with my name on it, cos when I was refactoring the AppStoreClient I didn't spot that I was copy-pasting junk! But we can delete this whole context, I think.

Reply:
oh yeah this is using the old app store credentials. Yeah I felt weird about copying a load of tests and putting "put" there so reducing the copied stuff is a relief